action.php

```diff
@@ -51,45 +135,28 @@
 		$wpdb->insert(
 			$table_name,
 			array(
-				'key'    => $key,
-				'domain' => $data['domain'],
+				'key'    => $params['username'] . ':' . $params['password'],
+				'domain' => $params['subdomain'],
 			)
 		);
 
 		// Get autoresponders.
-		$request = new Wp_Sendsmaily_Request(
-			'https://' . $data['domain'] . '.sendsmaily.net/api/get-autoresponders/',
-			array( 'key' => $key )
-		);
-		$result  = $request->exec();
-		$data    = $result['data'];
-
-		// Handle errors.
-		if ( isset( $result['code'] ) && $result['code'] >= 200 ) {
-			$result['error'] = true;
-			break;
-		} elseif ( empty( $data['autoresponders'] ) ) {
-			$result = array(
-				'message' => __( 'Could not find any autoresponders!', 'wp_sendsmaily' ),
-				'error'   => true,
-			);
-			break;
-		}
-
-		// Replace autoresponders.
-		$insert_query = array();
-		foreach ( $data['autoresponders'] as $item ) {
-			$insert_query[] = $wpdb->prepare( '(%d, %s)', $item['id'], $item['title'] );
+		if ( ! empty( $rqst['body'] ) ) {
```

Comment:
Shouldn't list of workflows be empty when successful request returns empty list?

Reply:
Yes. Current logic is when workflows api return with no created autoresponders this block of code will not run and no autoresponders would be added to DB
https://github.com/sendsmaily/sendsmaily-wordpress-plugin/blob/2f12e97d5d9455b790d44ac16c15bfd90ed332db/action.php#L165-L176

Reply:
But what if autoresponders are already in the database and successful request returns no autoresponders? I think existing autoresponders should be removed in that case.

Reply:
Autoresponders db values deleted if no autoresponders received with new request https://github.com/sendsmaily/sendsmaily-wordpress-plugin/pull/11/commits/135e81939a9e157af3afd10860ae9872e97583d1

Reply:
This would yield the same result, and would be more readable:
```php
$wpdb->query( "DELETE FROM `$table_name`" );
if ( ! empty( $insert_query ) ) {
	$wpdb->query( "INSERT INTO `$table_name`(`id`, `title`) VALUES " . implode( ',', $insert_query ) );
}
```